docs/index-pages/customizing-the-index-page.rst

```diff
@@ -360,8 +360,97 @@ must be created in the ``PostsTable`` class.
         }
     }
 
-Custom Download Links
+Download Format Links
 ---------------------
 
+The ``scaffold.index_formats`` configuration key can be used to customize
+"Download Links". These are alternative methods of displaying the current index
+page, and can be used to expose the paginated data in JSON, XML, or other
+formats. The output of each format can be customized to your specifications.
+
+The ``scaffold.index_formats`` option takes an array of download format data.
+Each sub-array should contain ``title`` and ``url`` parameters.
+
+.. code-block:: php
+
+    use Cake\Routing\Router;
+
+    // link to the current page, except with extensions `json` or `xml`
+    // include the querystring argument as specified or you will lose any
+    // currently applied filters
+    $action = $this->Crud->action();
+    $action->config('scaffold.index_formats', [
+        [
+            'title' => 'JSON',
+            'url' => ['_ext' => 'json', '?' => $this->request->query]
```

Comment:
Should we require CakePHP 3.4+? That was something I was iffy on in docs/functionality, but would remove a ton of deprecated calls.